code/modules/library/admin_only.dm

```diff
@@ -0,0 +1,354 @@
+			return TRUE
+
+/obj/machinery/computer/libraryconsole/admin_only_do_not_map_in_you_fucker/ui_data(mob/user)
+	. = ..()
+	.["view_raw"] = view_raw
+	.["show_deleted"] = show_deleted
+	var/list/histories = list()
+	for(var/id as anything in book_history)
+		var/list/insert = list()
+		for(var/datum/book_history_entry/entry in book_history[id])
+			insert += list(entry.serialize())
+		histories[id] = insert
+	.["history"] = histories
+
+/obj/machinery/computer/libraryconsole/admin_only_do_not_map_in_you_fucker/proc/view_book(id, mob/show_to)
+	if (!SSdbcore.Connect())
+		can_connect = FALSE
+		message_admins("Could not connect to the DB")
+		return
+
+	var/datum/db_query/query_library_view = SSdbcore.NewQuery(
+		"SELECT * FROM [format_table_name("library")] WHERE id=:id",
+		list("id" = id)
+	)
+	if(!query_library_view.Execute())
+		qdel(query_library_view)
+		return
+
+	while(query_library_view.NextRow())
+		var/datum/admin_book_viewer/viewer = new()
+		viewer.set_owner(src)
+		viewer.id = query_library_view.item[1]
+		viewer.author = query_library_view.item[2]
+		viewer.title = query_library_view.item[3]
+		viewer.content = query_library_view.item[4]
+		viewer.category = query_library_view.item[5]
+		viewer.author_ckey = query_library_view.item[6]
+		viewer.creation_time = query_library_view.item[7]
+		viewer.deleted = query_library_view.item[8]
+		viewer.creation_round = query_library_view.item[9]
+		viewer.history = get_book_history(id)
+		viewer.ui_interact(show_to)
+		break
+	qdel(query_library_view)
+
+/obj/machinery/computer/libraryconsole/admin_only_do_not_map_in_you_fucker/proc/get_book_history(id)
+	var/datum/db_query/query_book_history = SSdbcore.NewQuery({"
+		SELECT id, book, reason, ckey, datetime, action, INET_NTOA(ip_addr)
+			FROM [format_table_name("library_action")] WHERE book=:id
+		"},
+		list("id" = id)
+	)
+	if(!query_book_history.Execute())
+		qdel(query_book_history)
+		return list()
+
+	var/list/full_history = list()
+	while(query_book_history.NextRow())
+		var/datum/book_history_entry/history = new()
+		history.id = query_book_history.item[1]
+		history.book = query_book_history.item[2]
+		history.reason = query_book_history.item[3]
+		history.ckey = query_book_history.item[4]
+		history.datetime = query_book_history.item[5]
+		history.action = query_book_history.item[6]
+		history.ip_addr = query_book_history.item[7]
+		full_history += history
+	qdel(query_book_history)
+	return full_history
+
+/obj/machinery/computer/libraryconsole/admin_only_do_not_map_in_you_fucker/proc/hide_book(id, reason, client/admin)
+	if(!SSdbcore.Connect())
+		can_connect = FALSE
+		message_admins("Could not connect to the DB")
+		return
+	if(IsAdminAdvancedProcCall())
+		stack_trace("Proc call lead to hide_book being called, blocked")
+		message_admins("Proc call lead to hide_book being called, blocked")
+		return
+	if(!check_rights_for(admin, R_BAN))
+		stack_trace("[admin.ckey] tried to hide a book without the required perms")
```

Comment:
This is gonna be in the condensed runtime logs, seems like it should log confidentially rather than blare for everyone to see on runtime collaters and stuff like that.

Reply:
We'll use log_admin then